Code/Framework/AzCore/AzCore/Metrics/IEventLoggerFactory.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Outcome/Outcome.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/std/function/function_fwd.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+
+namespace AZ::Metrics
+{
+    class IEventLogger;
+
+    enum class EventLoggerId : AZ::u32 {};
+
+    AZ_DEFINE_ENUM_RELATIONAL_OPERATORS(EventLoggerId);
+
+    class IEventLoggerFactory
+    {
+    public:
+        AZ_RTTI(IEventLoggerFactory, "{3E98C565-3A1E-460E-B692-4FAE783952CC}");
+        virtual ~IEventLoggerFactory() = default;
+
+        //! Callback function that is invoked for every registered event logger
+        //! return true to continue visiting
+        using VisitEventLoggerInterfaceCallback = AZStd::function<bool(IEventLogger&)>;
+        //! Invokes the supplied visitor for each register compression interface that is non-nullptr
+        virtual void VisitEventLoggers(const VisitEventLoggerInterfaceCallback&) const = 0;
+
+        //! Registers event logger and takes ownership of it if registration is successful
+        //! @param eventLogger Event logger to register
+        //! @return true if the IEventLogger was successfully registered
+        //! If the logger was successfully registered a unique_ptr containing it is returned instead
```

Comment:
This isn't accurate. The return is an `Outcome` not a bool, and doesn't return a `unique_ptr` on success.  It would be better to mention that the `Outcome`'s `Failure` would contain the event logger and a success would transfer ownership of the logger to the internal system.

Reply:
Will fix.